File: backend/app/schemas/volcano.py
from typing import Literal

from pydantic import AwareDatetime, BaseModel, Field, HttpUrl


class Volcano(BaseModel):
    id: str
    name: str
    latitude: float | None = Field(default=None, ge=-90, le=90)
    longitude: float | None = Field(default=None, ge=-180, le=180)
    current_alert_level: int | None = Field(default=None, ge=0, le=5)
    source: Literal["phivolcs"] = "phivolcs"
    source_url: HttpUrl | None = None
    bulletin_url: HttpUrl | None = None
    bulletin_at: AwareDatetime | None = None
    retrieved_at: AwareDatetime | None = None
    stale: bool = False

File: web/src/types/volcano.ts
export type Volcano = {
  id: string;
  name: string;
  latitude: number | null;
  longitude: number | null;
  current_alert_level: number | null;
  source: 'phivolcs';
  source_url: string | null;
  bulletin_url: string | null;
  bulletin_at: string | null;
  retrieved_at: string | null;
  stale: boolean;
};
